OneSignalSDK/onesignal/src/main/java/com/onesignal/OSInAppMessageController.java

```diff
@@ -303,63 +308,42 @@ void onFailure(int statusCode, String response, Throwable throwable) {
         }
     }
 
-    private void messageCanBeDisplayed(@NonNull OSInAppMessage message) {
-        if (!inAppMessagingEnabled)
-            return;
-
-        if (triggeredMessages.contains(message.messageId) &&
-            !message.isPreview) {
-            OneSignal.Log(
-               OneSignal.LOG_LEVEL.ERROR,
-               "In-App message with id '" +
-                  message.messageId +
-                  "' already displayed or is already preparing to be display!");
-            return;
-        }
-
-        queueMessageForDisplay(message);
-    }
-
     // Message has passed triggers and de-duplication logic.
     // Display message now or add it to the queue to be displayed.
     private void queueMessageForDisplay(@NonNull OSInAppMessage message) {
         synchronized (messageDisplayQueue) {
             messageDisplayQueue.add(message);
```

Comment:
Need to check we don't add the same message more than once

Reply:
Got it